sdk/core/src/pipeline.rs

```diff
@@ -89,28 +89,26 @@ where
             pipeline.push(Arc::new(TransportPolicy::new(options.transport.clone())));
 
             #[cfg(feature = "mock_transport_framework")]
-            match std::env::var("TESTING_MODE") {
-                Ok(mode) => match mode.as_ref() {
-                    "RECORD" => {
-                        warn!("mock testing framework record mode enabled");
-                        pipeline.push(Arc::new(crate::policies::MockTransportRecorderPolicy::new(
-                            options.transport.clone(),
-                        )));
-                    }
-                    "PLAY" => {
-                        warn!("mock testing framework reply mode enabled");
-                        pipeline.push(Arc::new(crate::policies::MockTransportPlayerPolicy::new(
-                            options.transport.clone(),
-                        )));
-                    }
-                    _ => {
-                        warn!(
-                            "invalid TESTING_MODE selected. Supported options are PLAY and RECORD"
-                        );
-                        pipeline.push(Arc::new(TransportPolicy::new(options.transport.clone())));
-                    }
-                },
-                Err(_) => {} // ignore missing env variable and non-unicode ones
+            match std::env::var("TESTING_MODE").as_deref().unwrap_or("PLAY") {
+                "RECORD" => {
+                    info!("mock testing framework record mode enabled");
+                    pipeline.push(Arc::new(crate::policies::MockTransportRecorderPolicy::new(
+                        std::env::var("CARGO_BIN_NAME").unwrap(),
+                        options.transport.clone(),
+                    )));
+                }
+                "PLAY" => {
+                    info!("mock testing framework reply mode enabled");
+                    pipeline.push(Arc::new(crate::policies::MockTransportPlayerPolicy::new(
+                        std::env::var("CARGO_BIN_NAME")
+                            .unwrap_or("create_database_and_collection".into()),
```

Comment:
This is just hardcoded for the one test. We need to change how this works.

Reply:
I've pushed code now that passes the transaction name throught the transport options. It can be cleaned up a bit, but it works.